notification-lambda-cfn.yaml

```
@@ -90,4 +93,43 @@ Resources:
       FunctionName: !GetAtt Function.Arn
       Action: lambda:InvokeFunction
       Principal: events.amazonaws.com
-      SourceArn: !GetAtt EventRule.Arn
\ No newline at end of file
+      SourceArn: !GetAtt EventRule.Arn
+
+  LambdaAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmActions: [!Ref AlarmTopic]
+      OKActions: [!Ref AlarmTopic]
+      AlarmDescription: !Sub Alarm triggers if there is missing notification data, see runbook <<<Runbook|https://docs.google.com/document/d/1JQXFacHrAFB8_7fLthUkIdt1LPYTD99PxJOU_64rY2Y/edit?usp=sharing>>>
+      Metrics:
+        - Id: e1
+          Label: Error percentage of lambda
+          Expression: "100*(m1/m2)"
+        - Id: m1
+          Label: Number of errors for lambda
+          MetricStat:
+            Metric:
+              MetricName: Errors
+              Namespace: AWS/Lambda
+              Dimensions:
+                - Name: FunctionName
+                  Value: !Sub us-election-2020-notification-lambda-${Stage}
+            Period: 600
+            Stat: Sum
+          ReturnData: false
+        - Id: m2
+          Label: Number of invocations for lambda
+          MetricStat:
+            Metric:
+              MetricName: Invocations
+              Namespace: AWS/Lambda
+              Dimensions:
+                - Name: FunctionName
+                  Value: !Sub us-election-2020-notification-lambda-${Stage}
+            Period: 600
+            Stat: Sum
+          ReturnData: false
+      ComparisonOperator: GreaterThanOrEqualToThreshold
+      EvaluationPeriods: 1
+      Threshold: 1
```

```suggestion
      Threshold: 100
```

This is a percentage now. By choosing `100`, we should only receive an alert if the lambda has been consistently failing for 5 minutes (we can quickly test this in `CODE` by breaking the handler function). This is probably a good indicator that something has permanently broken and requires an action from whoever is on call.